scrapperFunctions/index.js

```diff
@@ -0,0 +1,227 @@
+    console.error(`Error: ${err}`);
+    throw err;
+  }
+};
+
+/**
+ * Click a button
+ * @param {string} id How identify the button
+ * @param {int | double} time Time to wait for the results
+ */
+// eslint-disable-next-line space-before-function-paren
+ScrapPage.prototype.clickButton = async function (id, time) {
+  try {
+    await this.page.click(id);
+    await this.page.waitForTimeout(time);
+  } catch (err) {
+    console.error(`Error: ${err}`);
+    throw err;
+  }
+};
+
+/**
+ *
+ * @param {string} selector identified for the table in DOM
+ * @returns {Array[Array[strings]]}  each array append is a row of the table,
+ * each element a cel
+ */
+
+const tableToArrays = (selector) => {
+  const selection = `${selector} tr`;
+  // Get the data with this selector
+  const elements = document.querySelectorAll(selection);
+  const inf = []; // Row set
+  for (let i = 0; i < elements.length; i++) {
+    const el = elements[i].children;
+    const indArr = []; // Data of each row
+    for (let j = 0; j < elements.length; j++) {
+      if (el[j].innerText == '' || el[j].innerText == 'undefined') {
+        indArr.push('---');
+      } else {
+        indArr.push(el[j].innerText);
+      }
+    }
+    inf.push(indArr);
+  }
+  return inf;
+  // eslint-disable-next-line comma-dangle
+};
+
+/**
+ * Get the number of rows in the table
+ * @param {string} selector Of where I'm going to get the data
+ * @returns {int | string} The data obtained
+ */
+
+const getRows = (selector) => {
+  const rows = document.querySelector(selector).innerText;
+  return rows;
+};
+
+/**
+ * How many rows there is
+ * @param {string} select How identify the element
+ * @returns {int | string} The data obtained
+ */
+// eslint-disable-next-line space-before-function-paren
+ScrapPage.prototype.expectedRows = async function (select) {
+  try {
+    const rowsQ = await this.page.evaluate(getRows, select);
+    return rowsQ;
+  } catch (err) {
+    return err;
+  }
+};
+
+/**
+ * Collect the data of different screens
+ * @param {string} tableSelector The identifier of the table
+ * @param {string} rowSelector How can I find the number of rows in the table
+ * @param {string} nextPageButton The identifier of the button
+ * @param {int | double} time How much time wait for do the process again
+ * @param {function} callback A function needed in progressBar
+ * @returns Array[Array[string]]
+ */
+// eslint-disable-next-line space-before-function-paren
+ScrapPage.prototype.checkData = async function (
+  tableSelector,
+  rowSelector,
+  nextPageButton,
+  time,
+  // eslint-disable-next-line prettier/prettier
+  callback,
+) {
+  try {
+    let exp = await this.expectedRows(rowSelector);
+    let data = await this.getDataTable(tableSelector);
+    const validateNumber = /^[0-9]*$/;
+    const onlyNumbers = validateNumber.test(exp);
+    typeof exp !== 'string' && onlyNumbers ? (exp = data.length) : exp;
+    let obt = data.length;
+    if (exp == 0) return exp;
```

Comment:
i saw there are a lot of usages for non-strict comparison, here's a good resource about what this means. Try to use strict comparison in all your ifs, it will give you typecheck

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality